howdoi/howdoi.py

```diff
@@ -621,7 +623,16 @@ def howdoi(raw_query):
         cache.set(cache_key, res)
     except (RequestsConnectionError, SSLError):
         res = {'error': f'Unable to reach {args["search_engine"]}. Do you need to use a proxy?\n'}
-
+    except BlockError:
+        BLOCKED_ENGINES.append(args['search_engine'])
+        if BLOCKED_ENGINES == SUPPORTED_SEARCH_ENGINES:
+            return howdoi(args)
+        for eng in SUPPORTED_SEARCH_ENGINES:
```

Comment:
Another way to write this could be 

```
[engine for engine in SUPPORTED_SEARCH_ENGINES if engine not in BLOCKED_ENGINES]
```

However, you then must reverse the list because if you started with ('google', 'bing', 'duckduckgo') you're going to get back ('duckduckgo', 'bing', 'google') and we probably want to try these search engines in the order they are written.

What do you think about:

```
next_search_engine = next(reversed([engine for engine in SUPPORTED_SEARCH_ENGINES if engine not in BLOCKED_ENGINES]), None)
```

You can check the result of this this, if it's `None` then you can print the error. If it's not then you can use that as the next search engine to try.

Reply:
Thanks for sharing this! I was looking for a more compact method to iterate through the list and find unblocked engines, one at a time, and this is perfect for that. I've added it in now, let me know what you think